In [75]:
import pandas as pd
import re

# 데이터 로드
sample_df = pd.read_csv('sample_data.csv')
sample_df12 = sample_df.copy()

# 지정된 카테고리 목록
valid_categories = [
    '국어-독서', '국어-문학', '국어-화법과작문', '국어-언어', '국어-매체',
    '사회-생활과윤리', '사회-윤리와사상', '사회-한국지리', '사회-세계지리',
    '사회-동아시아사', '사회-세계사', '사회-경제', '사회-정치와법', '사회-문화'
]

In [76]:
# 카테고리 열에서 숫자와 점 제거 (예: '1.', '2.' 등)
sample_df12['cleaned_category'] = sample_df12['category'].str.replace(r'^\d+\.\s*', '', regex=True)

# 지정된 카테고리에 해당하는 데이터 추출
standard_categories = sample_df12[sample_df12['cleaned_category'].isin(valid_categories)]

# 지정된 카테고리에 해당하지 않는 데이터 추출
non_standard_categories = sample_df12[~sample_df12['cleaned_category'].isin(valid_categories)]

In [77]:
len(standard_categories)


1912

In [78]:
len(non_standard_categories)

119

In [79]:
def clean_non_standard_category(category):
    # 줄바꿈 문자 제거 및 공백 정리
    category = re.sub(r'\s+', ' ', category).strip()
    
    # 접두어 제거
    prefixes_to_remove = ['과목:', '세부 주제:', '결과:', '카테고리:']
    for prefix in prefixes_to_remove:
        category = category.replace(prefix, '').strip()
    
    # 유사한 카테고리 통합 및 기타 정제 작업
    if category in ['심리-심리학', '과학-심리학']:
        return '과학-심리학'
    if category in ['사회-범죄와법', '사회-범죄와 법']:
        return '사회-범죄'
    if category in ['자연-생물학', '과학-생물', '과학-생물학']:
        return '과학-생물학'
    
    # 기타 정제 작업
    category = category.replace('사회-노동과 경제', '사회-경제')
    
    return category

In [80]:
# 비표준 카테고리에 대한 정제 함수 적용
non_standard_categories['cleaned_category'] = non_standard_categories['cleaned_category'].apply(clean_non_standard_category)

# None 값을 가진 행 제거
non_standard_categories_cleaned = non_standard_categories.dropna(subset=['cleaned_category'])

/tmp/ipykernel_1716899/43168141.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  non_standard_categories['cleaned_category'] = non_standard_categories['cleaned_category'].apply(clean_non_standard_category)


In [81]:
import re

def is_standard_category(category):
    return re.match(r'^(사회|수학|과학|국어)-', category) is not None

# 표준 카테고리 추출
ss_standard_categories = non_standard_categories_cleaned[non_standard_categories_cleaned['cleaned_category'].apply(is_standard_category)]

# 비표준 카테고리 추출
non_standard_categories = non_standard_categories_cleaned[~non_standard_categories_cleaned['cleaned_category'].apply(is_standard_category)]

In [82]:
filtered12 = pd.read_csv('filtered_data.csv')
sample_df123 = filtered12.copy()

In [83]:
sample_df123

id  \
0    generation-for-nlp-452   
1    generation-for-nlp-808   
2    generation-for-nlp-877   
3    generation-for-nlp-973   
4   generation-for-nlp-1496   
5   generation-for-nlp-1571   
6   generation-for-nlp-1920   
7   generation-for-nlp-2013   
8   generation-for-nlp-2070   
9   generation-for-nlp-2075   
10  generation-for-nlp-2229   
11  generation-for-nlp-2492   
12  generation-for-nlp-2779   
13  generation-for-nlp-2826   
14  generation-for-nlp-2882   

                                            paragraph  \
0   풍토에 따라 곡식을 심고 가꾸는 법이 다르니, 고을의 경험 많은 농부를 각 도의 감...   
1   멜라니는 매일 점심값으로 개당 1달러인 사과와 개당 2달러인 오렌지를 사 먹습니다....   
2   수면 연구에서 한 연구자는 해당 참가자의 눈이 스포츠 경기를 시청할 때와 같이 빠르...   
3   정서적 애착에 대한 할로우(Harlow)의 연구에서 새끼 원숭이들을 우리에 넣고 "...   
4   환경재단(이사장 최열)과 롯데홈쇼핑(대표이사 이완신)은 11월 10일(일) 문화비축...   
5   치매의 원인과 신경세포 사멸의 세포와 분자수준의 기전이 밝혀졌다. 기초과학연구원(원...   
6   직장인 김모씨(56·서울 문래동)는 지난 2월 딸을 서울의 한 대학 국제학부에 입학...   
7   이번 수능 영어는 1등급 비율이 7.43%였던 지난해 수능 대비 난이도가 비슷하거나...   
8   국내에서 B형 부모 사이에 AB형 자녀가 태어난 것으로 확인됐다. 돌연변이에 의한 ...   
9   POSTECH(포항공과대학교, 총장 김무환) 전자전기공학과 박부견 교수와 생명과학과...   
10  개인용 컴퓨터에 들어 있는 SRAM, DRAM은 높은 속도로 동작이 가능하지만 전원...   
11  식물이 숨을 쉴 때 내뿜는 수증기는 더운 날 마당에 물을 뿌리는 것과 같이 지면온도...   
12  서울시 산하 SH공사는 오는 18일부터 장기전세주택(시프트) 455가구를 공급한다고...   
13  터미널전자상가로 사용됐던 서울 용산관광버스터미널 부지에 2017년까지 대규모 관광호...   
14  혈액세포의 근간인 조혈줄기세포의 생성 원리가 밝혀졌다. 기초과학연구원(IBS, 원장...   

                                             question  \
0                               밑줄 친 ‘왕’의 업적으로 옳은 것은?   
1   멜라니가 점심값을 이미 다 썼다면 그녀가 효용을 극대화하기 위해 소비 결정에 있어서...   
2                  이 연구자는 참가자가 어느 단계에 진입했다고 기록해야 합니까?   
3                                 이 실험에서 독립변수는 다음입니다.   
4                         2019 에코페스트 인 서울의 슬로건은 무엇인가?   
5                     이번 연구에서 밝혀진 치매 유도의 핵심 요소는 무엇인가?   
6                          김모씨가 딸을 입학시킨 대학의 학부는 무엇인가?   
7        이번 수능 영어에서 1등급 비율이 지난해 대비 어떻게 변화할 것으로 전망되는가?   
8                      시스-AB형 혈액형이 발견된 여성의 나이는 몇 세인가?   
9   한국과학기술한림원의 정회원으로 선임된 두 교수 중 전자전기공학과 소속인 교수의 이름...   
10      POSTECH-서울대 공동연구팀이 발견한 Fe3GeTe2의 주요 특성은 무엇인가?   
11     이산화탄소 증가가 북극 온난화에 미치는 영향을 연구한 교수의 소속 대학은 어디인가?   
12               서울시 SH공사가 공급하는 장기전세주택의 공급 시작일은 언제인가?   
13           서울 용산관광버스터미널 부지에 건설될 관광호텔의 총 객실 수는 얼마인가?   
14        조혈줄기세포 발생에 핵심 역할을 하는 히스톤 샤페론 단백질의 이름은 무엇인가?   

                                              choices  answer  question_plus  \
0   ['공법을 제정하였다 .', '한양으로 도읍을 옮겼다 .', '경국대 전 을 완성하...       1            NaN   
1   ['그녀는 어떤 변화도 필요하지 않다. 사과와 오렌지의 효용을 극대화하는 조합을 소...       2            NaN   
2                  ['3단계 수면', '베타 수면', '렘수면', '비렘수면']       3            NaN   
3   ['우유병이 배치된 "엄마"', '"철사로 만든" 엄마 대 "헝겊으로 만든" 엄마'...       1            NaN   
4   ['지구에서 제대로 노는 법', '환경을 지키는 우리의 약속', '지속 가능한 삶의...       1            NaN   
5   ['아밀로이드베타', '중증 반응성 별세포', '모노아민 산화효소 B', '과산화수...       2            NaN   
6             ['의대', '한의대', '국제학부', '예체능 계열', '공과대학']       3            NaN   
7   ['비슷하거나 약간 높아질 것으로 전망된다.', '비슷하거나 약간 낮아질 것으로 전...       1            NaN   
8                 ['25세', '29세', '30세', '35세', '40세']       2            NaN   
9                 ['박부견', '이지오', '김무환', '정재훈', '이상엽']       1            NaN   
10  ['전원이 꺼져도 정보가 유지되는 강자성 물질이다.', '전류를 걸면 강자성 물질에...       2            NaN   
11  ['POSTECH(포항공과대학교)', '서울대학교', '고려대학교', '연세대학교'...       1            NaN   
12  ['4월 18일', '5월 1일', '3월 15일', '6월 10일', '7월 20일']       1            NaN   
13      ['1827실', '1500실', '2000실', '2500실', '3000실']       1            NaN   
14  ['섭티16에이치(Supt16h)', 'p53', 'Notch', 'phc1', '...       1            NaN   

                                   category cleaned_category  
0     사회의 역사적 사건에 대한 질문이므로, 해당 지문과 질문은 "사회-         사회-동아시아사  
1                  지문: 멜라니는 매일 점심값으로 개당 1달러            사회-경제  
2                  과목: 과학\n세부 주제: 심리학-수면 연구           과학-심리학  
3                    과목: 사회  \n세부 주제: 사회-문화            사회-문화  
4   주어진 지문과 질문을 분석한 결과, 해당 문제는 다음과 같이 분류됩니다            사회-환경  
5                과목: 과학  \n세부 주제: 생명과학-신경과학

In [84]:
standard_categories

id  \
1      generation-for-nlp-426   
2      generation-for-nlp-427   
3      generation-for-nlp-428   
4      generation-for-nlp-429   
5      generation-for-nlp-431   
...                       ...   
2026  generation-for-nlp-2893   
2027  generation-for-nlp-2894   
2028  generation-for-nlp-2895   
2029  generation-for-nlp-2896   
2030  generation-for-nlp-2899   

                                              paragraph  \
1     (가)은/는 의병계열과 애국계몽 운동 계열의 비밀결사가 모여 결성된 조직으로, 총사...   
2     나는 삼한(三韓) 산천의 음덕을 입어 대업을 이루었다.(가)는/은 수덕(水德)이 순...   
3     이 날 소정방이 부총관 김인문 등과 함께 기 벌포에 도착하여 백제 군사와 마주쳤다....   
4     선비들 수만 명이 대궐 앞에 모여 만 동묘와 서원을 다시 설립할 것을 청하니, (가...   
5     (가)의 사신 고제덕 등이 일본에 와서 왕이 보낸 국서를 전하였다. 그 국서에 이르...   
...                                                 ...   
2026  “헐값에 팔리는 냉동 오렌지주스만 잔뜩 사가고, 쿠폰을 오려 모았으며, 구멍 난 스...   
2027  넷기어코리아(지사장 김진겸, 이하 넷기어)가 뮤럴 캔버스를 가지고 넷기어 SNS에서...   
2028  서울 성동구 옥수동과 금호동은 맞붙어 있는 동네지만 아파트값은 같은 면적에서 1억원...   
2029  방하남 고용노동부 장관(사진)이 취임 후 첫 외부 행사로 5년 만에 일자리를 3.5...   
2030  진실한 사랑은 기쁨과 평화를 낳고 고통을 줄여준다. 그것은 자애, 연민, 기쁨, 평...   

                                  question  \
1                   (가)에 대한 설명으로 옳지 않은 것은?   
2                   (가) 지역에 대한 설명으로 옳은 것은?   
3                 밑줄 친 ‘그’에 대한 설명으로 옳은 것은?   
4               (가) 인물이 추진한 정책으로 옳지 않은 것은?   
5                   (가) 국가에 대한 설명으로 옳은 것은?   
...                                    ...   
2026              잭 맥도널드가 남긴 유산의 총액은 얼마인가?   
2027   넷기어가 모집하는 '뮤럴 공식 도슨트'의 주요 역할은 무엇인가?   
2028   옥수동과 금호동의 아파트값 차이가 발생한 주된 이유는 무엇인가?   
2029       방하남 고용노동부 장관이 방문한 기업의 이름은 무엇인가?   
2030  틱낫한 스님이 강조하는 진실한 사랑의 요소가 아닌 것은 무엇인가?   

                                                choices  answer  \
1     ['고려 문종 때에 남경(南京)으로 승격되었다.', '종루(鐘樓), 이현, 칠패 등...       1   
2     ['이곳에 대장도감을 설치하여 재조대장경을 만들었다.', '지눌이 이곳에서 수선사 ...       4   
3     ['살수에서 수의 군대를 물리쳤다 .', '김춘추 의 신라 왕위 계승을 지원하였다 ...       2   
4     ['사창제를 실시하였다 .', '대전회통을 편찬하였다 .', '비변사의 기능을 강화...       3   
5     ['나 당연합군의 공격으로 멸망하였다 .', '9주 5소경의 지방 행정 구역을 두었...       3   
...                                                 ...     ...   
2026   ['1억8760만달러', '5000만달러', '2억달러', '1억달러', '3억달러']       1   
2027  ['전시물 설명 및 안내', '고객 서비스 제공', '제품 판매 촉진', '온라인 ...       4   
2028  ['옥수동의 재개발이 먼저 이루어졌다', '금호동의 인구가 더 많다', '옥수동이 ...       1   
2029          ['셀트리온', '삼성전자', 'LG화학', '한화그룹', '현대자동차']       1   
2030                    ['자애', '연민', '기쁨', '평온', '소유욕']       5   

      question_plus  category cleaned_category  
1               NaN   사회-한국지리          사회-한국지리  
2               NaN  사회-동아시아사         사회-동아시아사  
3               NaN  사회-동아시아사         사회-동아시아사  
4               NaN   사회-한국지리          사회-한국지리  
5               NaN  사회-동아시아사         사회-동아시아사  
...             ...       ...              ...  
2026            NaN     사회-경제            사회-경제  
2027            NaN     사회-문화            사회-문화  
2028            NaN     사회-경제            사회-경제  
2029            NaN     사회-경제            사회-경제  
2030            NaN  사회-윤리와사상         사회-윤리와사상  

[1912 rows x 8 columns]

In [85]:
ss_standard_categories

id  \
0      generation-for-nlp-425   
34     generation-for-nlp-465   
54     generation-for-nlp-491   
86     generation-for-nlp-529   
111    generation-for-nlp-557   
...                       ...   
1987  generation-for-nlp-2848   
1988  generation-for-nlp-2849   
1990  generation-for-nlp-2851   
1996  generation-for-nlp-2857   
2012  generation-for-nlp-2875   

                                              paragraph  \
0     상소하여 아뢰기를 , “신이 좌참 찬 송준길이 올린 차자를 보았는데 , 상복(喪服)...   
34    (가) 김시민이 진주성에서 일본군을 저지하였다.(나) 조선수군이 명량해전에서 크게 ...   
54    왕이 복주에 이르렀다.정세운은 성품이 충성스럽고 청렴하였는데, 왕의 파천(播遷) 이...   
86    먼저 우주가 구형이라는 점에 주목해야 한다. 부분적으로는 연결점이 필요 없는 완전한...   
111   1848년은 독일과 유럽 역사에 있어 결정적인 해로, 독일의 과거를 요약하고 독일의...   
...                                                 ...   
1987  서울교육청이 내년부터 서울시내 초·중·고등학교의 9시 등교를 추진한다.조희연 서울교...   
1988  보이스피싱 사기 현행범이 서울의 한 새마을금고에서 범죄 수익금을 인출하려다 새마을금...   
1990  철도부지 등에 건설되는 공공임대주택인 ‘행복주택’이 주거·호텔·상업·업무시설 등이 ...   
1996  미국에 이어 스페인에서도 에볼라 바이러스에 감염된 환자가 확인됐다.월스트리트저널(W...   
2012  롯데건설이 서울 금천구 독산동에 건설 중인 ‘롯데캐슬 골드파크’는 도심에 들어서는 ...   

                                               question  \
0                 상소한 인물이 속한 붕당에 대한 설명으로 옳은 것만을 모두 고르면?   
34                      임진왜란의 주요 사건을 시기 순으로 바르게 나열한 것은?   
54                       밑줄 친 ‘왕’의 재위 기간에 있었던 일로 옳은 것은?   
86    위 글을 보았을 때 구체 형태의 우주에 관한 코페르니쿠스의 주장은 무엇을 근거로 하...   
111   다음 중 인용문에서 테일러가 1848년 정치 혁명이 끼친 가장 중요한 영향이라고 주...   
...                                                 ...   
1987                    서울교육청이 추진하는 9시 등교의 주된 목적은 무엇인가?   
1988                        보이스피싱 사기 현행범이 체포된 장소는 어디인가?   
1990                     행복주택의 주거환경 개선을 통한 주요 목적은 무엇인가?   
1996         스페인에서 에볼라 바이러스에 감염된 간호사가 치료한 환자의 이름은 무엇인가?   
2012              롯데캐슬 골드파크의 주요 특징 중 하나로 옳지 않은 것은 무엇인가?   

                                                choices  answer  \
0                      ['ㄱ, ㄴ', 'ㄱ, ㄷ', 'ㄴ, ㄹ', 'ㄷ, ㄹ']       2   
34    ['(가)→(다)→(마)→(라)→(나)', '(가)→(마)→(다)→(나)→(라)',...       4   
54    ['이제현에 의해 사략이 편찬되었다.', '직지 심체요절이 금속활자로 인쇄되었다.'...       1   
86         ['관찰과 귀납', '과거 기록의 권위', '실험', '제1원칙으로부터 추론']       1   
111   ['지배계급 변화에 실패', '사상의 힘을 보여줌', '보수주의와 자유주의 정치이념...       3   
...                                                 ...     ...   
1987  ['학생들의 수면 시간을 늘리기 위해', '사교육을 장려하기 위해', '학교 시설을...       1   
1988  ['서울 사당동 동작새마을금고', '서울 강남구 하나은행', '서울 종로구 국민은행...       1   
1990  ['슬럼화 방지와 일자리 창출', '주거비 절감과 세금 인하', '주택 공급 증가와...       1   
1996  ['마누엘 가르시아 비에호', '토머스 에릭 던컨', '시에라리온', '월스트리트저...       1   
2012  ['주거·상업·호텔·문화·공공·교육시설이 모두 갖춰진다.', '단지 안에 대형 공원...       4   

      question_plus     category cleaned_category  
0               NaN       사회-한국사           사회-한국사  
34              NaN       사회-한국사           사회-한국사  
54              NaN       사회-한국사           사회-한국사  
86              NaN      사회-자연과학          사회-자연과학  
111             NaN       사회-유럽사           사회-유럽사  
...             ...          ...              ...  
1987            NaN        사회-교육            사회-교육  
1988            NaN     사회-범죄와 법            사회-범죄  
1990            NaN   사회-주택과사회복지       사회-주택과사회복지  
1996            NaN      사회-생명과학          사회-생명과학  
2012            NaN  카테고리: 사회-경제            사회-경제  

[104 rows x 8 columns]

In [86]:
combined_df = pd.concat([sample_df123, standard_categories, ss_standard_categories], ignore_index=True)



In [87]:
combined_df

id  \
0      generation-for-nlp-452   
1      generation-for-nlp-808   
2      generation-for-nlp-877   
3      generation-for-nlp-973   
4     generation-for-nlp-1496   
...                       ...   
2026  generation-for-nlp-2848   
2027  generation-for-nlp-2849   
2028  generation-for-nlp-2851   
2029  generation-for-nlp-2857   
2030  generation-for-nlp-2875   

                                              paragraph  \
0     풍토에 따라 곡식을 심고 가꾸는 법이 다르니, 고을의 경험 많은 농부를 각 도의 감...   
1     멜라니는 매일 점심값으로 개당 1달러인 사과와 개당 2달러인 오렌지를 사 먹습니다....   
2     수면 연구에서 한 연구자는 해당 참가자의 눈이 스포츠 경기를 시청할 때와 같이 빠르...   
3     정서적 애착에 대한 할로우(Harlow)의 연구에서 새끼 원숭이들을 우리에 넣고 "...   
4     환경재단(이사장 최열)과 롯데홈쇼핑(대표이사 이완신)은 11월 10일(일) 문화비축...   
...                                                 ...   
2026  서울교육청이 내년부터 서울시내 초·중·고등학교의 9시 등교를 추진한다.조희연 서울교...   
2027  보이스피싱 사기 현행범이 서울의 한 새마을금고에서 범죄 수익금을 인출하려다 새마을금...   
2028  철도부지 등에 건설되는 공공임대주택인 ‘행복주택’이 주거·호텔·상업·업무시설 등이 ...   
2029  미국에 이어 스페인에서도 에볼라 바이러스에 감염된 환자가 확인됐다.월스트리트저널(W...   
2030  롯데건설이 서울 금천구 독산동에 건설 중인 ‘롯데캐슬 골드파크’는 도심에 들어서는 ...   

                                               question  \
0                                 밑줄 친 ‘왕’의 업적으로 옳은 것은?   
1     멜라니가 점심값을 이미 다 썼다면 그녀가 효용을 극대화하기 위해 소비 결정에 있어서...   
2                    이 연구자는 참가자가 어느 단계에 진입했다고 기록해야 합니까?   
3                                   이 실험에서 독립변수는 다음입니다.   
4                           2019 에코페스트 인 서울의 슬로건은 무엇인가?   
...                                                 ...   
2026                    서울교육청이 추진하는 9시 등교의 주된 목적은 무엇인가?   
2027                        보이스피싱 사기 현행범이 체포된 장소는 어디인가?   
2028                     행복주택의 주거환경 개선을 통한 주요 목적은 무엇인가?   
2029         스페인에서 에볼라 바이러스에 감염된 간호사가 치료한 환자의 이름은 무엇인가?   
2030              롯데캐슬 골드파크의 주요 특징 중 하나로 옳지 않은 것은 무엇인가?   

                                                choices  answer  \
0     ['공법을 제정하였다 .', '한양으로 도읍을 옮겼다 .', '경국대 전 을 완성하...       1   
1     ['그녀는 어떤 변화도 필요하지 않다. 사과와 오렌지의 효용을 극대화하는 조합을 소...       2   
2                    ['3단계 수면', '베타 수면', '렘수면', '비렘수면']       3   
3     ['우유병이 배치된 "엄마"', '"철사로 만든" 엄마 대 "헝겊으로 만든" 엄마'...       1   
4     ['지구에서 제대로 노는 법', '환경을 지키는 우리의 약속', '지속 가능한 삶의...       1   
...                                                 ...     ...   
2026  ['학생들의 수면 시간을 늘리기 위해', '사교육을 장려하기 위해', '학교 시설을...       1   
2027  ['서울 사당동 동작새마을금고', '서울 강남구 하나은행', '서울 종로구 국민은행...       1   
2028  ['슬럼화 방지와 일자리 창출', '주거비 절감과 세금 인하', '주택 공급 증가와...       1   
2029  ['마누엘 가르시아 비에호', '토머스 에릭 던컨', '시에라리온', '월스트리트저...       1   
2030  ['주거·상업·호텔·문화·공공·교육시설이 모두 갖춰진다.', '단지 안에 대형 공원...       4   

      question_plus                                 category cleaned_category  
0               NaN    사회의 역사적 사건에 대한 질문이므로, 해당 지문과 질문은 "사회-         사회-동아시아사  
1               NaN                 지문: 멜라니는 매일 점심값으로 개당 1달러            사회-경제  
2               NaN                 과목: 과학\n세부 주제: 심리학-수면 연구           과학-심리학  
3               NaN                   과목: 사회  \n세부 주제: 사회-문화            사회-문화  
4               NaN  주어진 지문과 질문을 분석한 결과, 해당 문제는 다음과 같이 분류됩니다            사회-환경  
...             ...                                      ...              ...  
2026            NaN                                    사회-교육            사회-교육  
2027            NaN                                 사회-범죄와 법            사회-범죄  
2028            NaN                               사회-주택과사회복지       사회-주택과사회복지  
2029            NaN                                  사회-생명과학          사회-생명과학  
2030            NaN                              카테고리: 사회-경제            사회-경제  

[2031 rows x 8 columns]

In [88]:
# id에서 숫자 부분만 추출하여 새로운 열에 저장
combined_df['numeric_id'] = combined_df['id'].apply(lambda x: int(x.split('-')[-1]))

# numeric_id를 기준으로 정렬
combined_df_sorted = combined_df.sort_values(by='numeric_id')

In [89]:
combined_df_sorted

id  \
1927   generation-for-nlp-425   
15     generation-for-nlp-426   
16     generation-for-nlp-427   
17     generation-for-nlp-428   
18     generation-for-nlp-429   
...                       ...   
1922  generation-for-nlp-2893   
1923  generation-for-nlp-2894   
1924  generation-for-nlp-2895   
1925  generation-for-nlp-2896   
1926  generation-for-nlp-2899   

                                              paragraph  \
1927  상소하여 아뢰기를 , “신이 좌참 찬 송준길이 올린 차자를 보았는데 , 상복(喪服)...   
15    (가)은/는 의병계열과 애국계몽 운동 계열의 비밀결사가 모여 결성된 조직으로, 총사...   
16    나는 삼한(三韓) 산천의 음덕을 입어 대업을 이루었다.(가)는/은 수덕(水德)이 순...   
17    이 날 소정방이 부총관 김인문 등과 함께 기 벌포에 도착하여 백제 군사와 마주쳤다....   
18    선비들 수만 명이 대궐 앞에 모여 만 동묘와 서원을 다시 설립할 것을 청하니, (가...   
...                                                 ...   
1922  “헐값에 팔리는 냉동 오렌지주스만 잔뜩 사가고, 쿠폰을 오려 모았으며, 구멍 난 스...   
1923  넷기어코리아(지사장 김진겸, 이하 넷기어)가 뮤럴 캔버스를 가지고 넷기어 SNS에서...   
1924  서울 성동구 옥수동과 금호동은 맞붙어 있는 동네지만 아파트값은 같은 면적에서 1억원...   
1925  방하남 고용노동부 장관(사진)이 취임 후 첫 외부 행사로 5년 만에 일자리를 3.5...   
1926  진실한 사랑은 기쁨과 평화를 낳고 고통을 줄여준다. 그것은 자애, 연민, 기쁨, 평...   

                                   question  \
1927  상소한 인물이 속한 붕당에 대한 설명으로 옳은 것만을 모두 고르면?   
15                   (가)에 대한 설명으로 옳지 않은 것은?   
16                   (가) 지역에 대한 설명으로 옳은 것은?   
17                 밑줄 친 ‘그’에 대한 설명으로 옳은 것은?   
18               (가) 인물이 추진한 정책으로 옳지 않은 것은?   
...                                     ...   
1922               잭 맥도널드가 남긴 유산의 총액은 얼마인가?   
1923    넷기어가 모집하는 '뮤럴 공식 도슨트'의 주요 역할은 무엇인가?   
1924    옥수동과 금호동의 아파트값 차이가 발생한 주된 이유는 무엇인가?   
1925        방하남 고용노동부 장관이 방문한 기업의 이름은 무엇인가?   
1926   틱낫한 스님이 강조하는 진실한 사랑의 요소가 아닌 것은 무엇인가?   

                                                choices  answer  \
1927                   ['ㄱ, ㄴ', 'ㄱ, ㄷ', 'ㄴ, ㄹ', 'ㄷ, ㄹ']       2   
15    ['고려 문종 때에 남경(南京)으로 승격되었다.', '종루(鐘樓), 이현, 칠패 등...       1   
16    ['이곳에 대장도감을 설치하여 재조대장경을 만들었다.', '지눌이 이곳에서 수선사 ...       4   
17    ['살수에서 수의 군대를 물리쳤다 .', '김춘추 의 신라 왕위 계승을 지원하였다 ...       2   
18    ['사창제를 실시하였다 .', '대전회통을 편찬하였다 .', '비변사의 기능을 강화...       3   
...                                                 ...     ...   
1922   ['1억8760만달러', '5000만달러', '2억달러', '1억달러', '3억달러']       1   
1923  ['전시물 설명 및 안내', '고객 서비스 제공', '제품 판매 촉진', '온라인 ...       4   
1924  ['옥수동의 재개발이 먼저 이루어졌다', '금호동의 인구가 더 많다', '옥수동이 ...       1   
1925          ['셀트리온', '삼성전자', 'LG화학', '한화그룹', '현대자동차']       1   
1926                    ['자애', '연민', '기쁨', '평온', '소유욕']       5   

      question_plus  category cleaned_category  numeric_id  
1927            NaN    사회-한국사           사회-한국사         425  
15              NaN   사회-한국지리          사회-한국지리         426  
16              NaN  사회-동아시아사         사회-동아시아사         427  
17              NaN  사회-동아시아사         사회-동아시아사         428  
18              NaN   사회-한국지리          사회-한국지리         429  
...             ...       ...              ...         ...  
1922            NaN     사회-경제            사회-경제        2893  
1923            NaN     사회-문화            사회-문화        2894  
1924            NaN     사회-경제            사회-경제        2895  
1925            NaN     사회-경제            사회-경제        2896  
1926            NaN  사회-윤리와사상         사회-윤리와사상        2899  

[2031 rows x 9 columns]

In [90]:
combined_df_sorted.to_csv('combined_sorted_data.csv', index=False)

In [91]:
# cleaned_category 열의 값 개수 계산
category_counts = combined_df_sorted['cleaned_category'].value_counts()

# 전체 개수 계산
total_count = len(combined_df_sorted)

# 비율 계산
category_percentages = (category_counts / total_count) * 100

# 결과 출력
print("각 카테고리의 비율:")
for category, percentage in category_percentages.items():
    print(f"{category}: {percentage:.2f}%")

각 카테고리의 비율:
사회-경제: 39.14%
사회-정치와법: 19.69%
사회-문화: 14.77%
사회-윤리와사상: 6.15%
사회-세계사: 4.38%
국어-문학: 2.95%
국어-독서: 1.82%
사회-생활과윤리: 1.58%
사회-한국지리: 1.58%
사회-동아시아사: 1.48%
사회-교육: 1.18%
국어-언어: 0.79%
사회-심리학: 0.54%
사회-환경: 0.44%
과학-생물학: 0.25%
과학-심리학: 0.25%
사회-한국사: 0.20%
수학-통계: 0.15%
국어-화법과작문: 0.15%
국어-매체: 0.10%
과학-생명과학: 0.10%
사회-과학: 0.10%
사회-의학: 0.10%
사회-범죄: 0.10%
과학-지구과학: 0.10%
사회-보건: 0.10%
사회-세계지리: 0.10%
사회-사회문제: 0.10%
사회-자연과학: 0.05%
사회-정신건강: 0.05%
과학-신경과학: 0.05%
사회-유럽사: 0.05%
사회-연구방법론: 0.05%
사회-인구와사회: 0.05%
과학-천문학: 0.05%
사회-주거와부동산: 0.05%
사회-역사: 0.05%
과학-기술: 0.05%
사회-스포츠: 0.05%
사회-보건과안전: 0.05%
과학-기상학: 0.05%
사회-주거와도시개발: 0.05%
사회-사건과재난: 0.05%
사회-주택 정책: 0.05%
사회-환경과기후변화: 0.05%
과학-기후 및 환경 과학: 0.05%
사회-기술과사회: 0.05%
사회-교통과인프라: 0.05%
과학-물리학: 0.05%
사회-교통과환경: 0.05%
사회-범죄와사회: 0.05%
사회-사고와재난: 0.05%
과학-화학: 0.05%
과학-환경과학: 0.05%
사회-정보 및 사이버 보안: 0.05%
사회-사고 및 재난 관리: 0.05%
사회-건강과환경: 0.05%
사회-의료와학: 0.05%
사회-환경과 지속가능성: 0.05%
사회-주택과사회복지: 0.05%
사회-생명과학: 0.05%


## 국어 영역 (총 5.81%)
- **국어-문학**: 2.95%
- **국어-독서**: 1.82%
- **국어-언어**: 0.79%
- **국어-화법과작문**: 0.15%
- **국어-매체**: 0.10%

## 사회탐구 영역 (총 91.89%)
- **사회-경제**: 39.14%
- **사회-정치와법**: 19.69%
- **사회-문화**: 14.77%
- **사회-윤리와사상**: 6.15%
- **사회-세계사**: 4.38%
- **사회-생활과윤리**: 1.58%
- **사회-한국지리**: 1.58%
- **사회-동아시아사**: 1.48%
- **사회-교육**: 1.18%
- 기타 사회 관련 주제: 1.94% (예: 사회-심리학, 사회-환경, 사회-한국사 등)

## 과학 영역 (총 0.70%)
- **과학-생물학**: 0.25%
- **과학-심리학**: 0.25%
- **과학-생명과학**: 0.10%
- **과학-지구과학**: 0.10%

## 수학 영역 (총 0.15%)
- **수학-통계**: 0.15%